pipelines/batch/setup_job.py

```diff
@@ -363,6 +463,46 @@ def main(
         default="pyrenew",
     )
 
+    # Function to convert string to boolean
+    # This is used to allow passing boolean values as command line arguments
+    # Reference: https://docs.python.org/3/library/argparse.html#type
+    # and
+    # https://stackoverflow.com/a/43357954
+    def string_to_boolean(value: str | bool) -> bool:
+        if isinstance(value, bool):
+            return value
+        if value.lower() in ("yes", "true", "t", "y", "1"):
+            return True
+        elif value.lower() in ("no", "false", "f", "n", "0"):
+            return False
+        else:
+            raise argparse.ArgumentTypeError("Boolean value expected.")
+
+    # With the string_to_boolean argparse type arg,
+    # you can supply the flag for True, omit it for False, or pass True/False explicitly
+    parser.add_argument(
+        "--test",
+        type=string_to_boolean,
+        nargs="?",
+        const=True,
+        default=False,
+        help="Run in test mode (default: False). Pass --test True or --test False to set explicitly.",
+    )
+
+    # With the string_to_boolean argparse type arg,
+    # you can supply the flag for True, omit it for False, or pass True/False explicitly
+    parser.add_argument(
+        "--dry-run",
+        type=string_to_boolean,
+        nargs="?",
+        const=True,
+        default=False,
+        help=(
+            "If set to True, do not submit tasks to Azure Batch. Only print what would be done."
+            "Pass --dry-run True or --dry-run False to set explicitly; omit to use default (False)."
+        ),
+    )
+
```

Comment:
Seems a bit complicated.
```python
parser.add_argument(
    "--test", action="store_true", default=False, help="blah"
)
```

makes it so `python myscript.py --test` results in the `test` arg being `True`, while `python myscript.py` results in the `test` arg being `False`. Seems much more ergonomic for anyone running the command line script directly.

I guess it would be hard for the make file to interpret `make run_e_model TEST=No` correctly.